### ISSUE TITLE
[NEW] How about support Roaring Bitmaps in valkey

### ISSUE THREAD

**Issue by im-bravo** [madolson/placeholderkv#1480]
**The problem/use-case that the feature addresses**
I want store large number of user id whitelist (10,000,000 ~ 50,000,000 ) with using bitmap and the max value of the user ID will be 2,147,483,647.
If there is only 1 user  in the bitmap and the value is 2,147,483,647, it will spend around 320MB memory.
If we can use some compression technology to reduce memory usage without affecting performance or even improve some performance, I think that would definitely be a great thing.
Maybe roaring bitmap is good choose to do some improvement.

**Description of the feature**
Support Roaring Bitmaps with new Commands

**Alternatives you've considered**
Support Roaring Bitmaps with existing bitmap commands if bitmap_engine="roaring" in config file.
bitmap_engine="string" as the default config to keep existing behavior.

**Additional information**

About Roaring Bitmap
https://roaringbitmap.org/
https://github.com/RoaringBitmap/CRoaring
Some implementations from others
https://github.com/aviggiano/redis-roaring

**Comment by im-bravo** [comment]
For java client, user may can integrate with Jedis to define some custom commands ?
https://github.com/redis/jedis?tab=readme-ov-file#using-redis-modules

**Comment by arthurkiller** [comment]
croaring is under Apache license and redis-roaring is MIT, you can use them with valkey freely

**Comment by arthurkiller** [comment]
and if you look a little deeply into roaring implementation, you will find roaring can adaptively choose `bitmap_engine` based on your data scale

**Comment by fjmaco** [comment]
https://github.com/fjmaco/valkey-roaring

**Comment by yzqzss** [comment]
https://github.com/aviggiano/redis-roaring is essentially a Redis distribution that includes the `redis-roaring` module. You can actually build the module separately and use it with Valkey. I have done this myself without encountering any issues: https://github.com/yzqzss/redis-roaring-builder

**Comment by fjmaco** [comment]
@yzqzss Yep, I know redis-roaring well — I've used it, and you're right that the module builds and loads into Valkey without issues (nice builder repo!). valkey-roaring exists for a different reason: it's a ground-up Rust rewrite on roaring-rs and Valkey's official Rust SDK, command-compatible with redis-roaring, plus one addition redis-roaring doesn't have. R.EXPORT/R.IMPORT for raw binary serialization in the CRoaring portable format. My main use case is microservice communication: a service asks for a large membership set and instead of an integer-array reply, it gets the compressed bitmap itself — typically an order of magnitude smaller on the wire — and deserializes it natively with any Roaring library (Java/Go/Python/C++/Rust). Low bandwidth, low memory on the receiving side, no parsing step.
